### ISSUE TITLE
Gallery linked caption text is barely visible

### ISSUE THREAD

**Issue by afercia** [WordPress/twentytwentyone#750]
**Describe the bug**
Noticed while testing the issue on #728.

When gallery images have a linked caption text, the link text is barely visible regardless whether the image is predominantly dark or light. The caption area uses a dark linear gradient and the link text is dark, thus the link doesn't have a sufficient contrast or it's even almost invisible.

Front end:

<img width="658" alt="21" src="https://user-images.githubusercontent.com/1682452/97682595-7712bd00-1a98-11eb-9a04-a9d1eb2bbb94.png">

Within the editor:

<img width="677" alt="Screenshot 2020-10-30 at 10 03 57" src="https://user-images.githubusercontent.com/1682452/97682401-69f5ce00-1a98-11eb-8980-c64dfc5fcf87.png">

Comparison with Twenty Twenty:

<img width="636" alt="20" src="https://user-images.githubusercontent.com/1682452/97682316-63675680-1a98-11eb-8382-c638b4fac14f.png">

**To Reproduce**
- add a gallery block with two images: a dark image and a light image 
- add caption text and link it
- observe the link text color becomes a dark grey
- same on hover and focus

**Expected behavior**
The linked caption text to be clearly visible.
